packages/stack/proxy/src/proxy.js

```diff
@@ -18,41 +18,46 @@ export class Proxy {
     this.logger = options.logger;
     this.vms = options.vms;
     this.app = null;
-    this.requestManager;
+    this.endpoint = options.endpoint;
+    if (options.endpoint === constants.blockchain.vm) {
+      this.endpoint = this.vms[this.vms.length - 1]();
+    }
+    this.isWs = options.isWs;
   }
 
-  async serve(endpoint, localHost, localPort, ws) {
-    if (endpoint === constants.blockchain.vm) {
-      endpoint = this.vms[this.vms.length - 1]();
-    }
-    this.requestManager = new Web3RequestManager.Manager(endpoint);
+  async nodeReady() {
+    const requestManager = new Web3RequestManager.Manager(this.endpoint);
 
     try {
-      await this.requestManager.send({ method: 'eth_accounts' });
+      await requestManager.send({ method: 'eth_accounts' });
     } catch (e) {
-      throw new Error(__('Unable to connect to the blockchain endpoint'));
+      throw new Error(__(`Unable to connect to the blockchain endpoint on ${this.endpoint}`));
     }
 
+    // disconnect socket connection (NO-OP when HTTP)
+    requestManager.provider.disconnect();
+  }
+
+  async serve(localHost, localPort) {
+
+    await this.nodeReady();
```

Comment:
Given that this can fail, we probably want to wrap some error handling around this. Given how `nodeReady` behaves, it probably makes sense to propagate the underlying error and have it output to the user somewhere in the CLI interface.

Reply:
This error will bubble up to the `setupProxy` method in `index.ts`, so have wrapped the call to `.setupProxy()` with a `try/catch` to output the underlying error (https://github.com/embark-framework/embark/pull/1983/files#diff-6baae298925cf40f841cd1ffc8175935R29).

Reply:
Ah sweet! Thanks for pointing that out